visallo/visallo
# Don't Use CompositeQuery when performing a "Search Related"

CompositeQuery doesn't properly handle sorting and computing the total hits when performing a "Search Related" query. This update switched to using the new `hasId` query method in Vertexium to retrieve all of the matching vertices in a single query.


Testing Instructions:
1. Right click on a vertex and select "Search -> Related".
1. Ensure that all vertices related to the clicked vertex are returned.
1. Add a `Relationship Type` filter and ensure that only the matching vertices are returned
1. Add another `Relationship Type` filter that does not match any vertices and ensure that the search results are the same as the previous step.
1. Remove the original `Relationship Type` filter and ensure that no results are found.
1. Select multiple vertices, right click, select "Search -> Related", and repeat the steps above.

CHANGELOG
Fixed: Sorting search results when performing a "Search Related" query.
